File: src/main/java/com/ktDemo/infrastructure/doma/entity/UsersEntity.java
package com.ktDemo.infrastructure.doma.entity;

import org.seasar.doma.Entity;
import org.seasar.doma.Id;
import org.seasar.doma.Table;
import org.seasar.doma.jdbc.entity.NamingType;

/**
 *
 */
@Entity(naming = NamingType.SNAKE_LOWER_CASE)
@Table(name = "users")
public class UsersEntity {

    /**  */
    @Id
    public Integer id;

    /**  */
    public String firstName;

    /**  */
    public String lastName;
}

File: src/main/kotlin/com/ktDemo/infrastructure/hibernate/entity/UsersEntity.kt
package com.ktDemo.infrastructure.hibernate.entity

import java.io.Serializable
import javax.persistence.*


/**
 * Users Entity
 */
@Entity
@Table(name = "users")
data class UsersEntity(

        /** ID */
        @Id
        @Column(name = "id", nullable = false)
        var id: Long = -1,

        /** 性  */
        @Column(name = "first_name", nullable = false)
        var firstName: String = "",

        /** 名  */
        @Column(name = "last_name", nullable = true)
        var lastName: String? = null,

        @OneToOne(cascade = arrayOf(CascadeType.ALL))
        val detail: DetailsEntity? = null

) : Serializable